homeassistant/components/acmeda/base.py

```diff
@@ -0,0 +1,58 @@
+"""Base class for Acmeda Roller Blinds."""
+import aiopulse
+
+from homeassistant.helpers import entity
+
+from .const import DOMAIN, LOGGER
+
+
+class AcmedaBase(entity.Entity):
+    """Base representation of an Acmeda roller."""
+
+    def __init__(self, hass, roller: aiopulse.Roller):
+        """Initialize the roller."""
+        self.hass = hass
+        self.roller = roller
+
+    async def async_added_to_hass(self):
+        """Entity has been added to hass."""
+        self.roller.callback_subscribe(self.notify_update)
+
+    async def async_reset(self):
+        """Entity being removed from hass."""
+        self.roller.callback_unsubscribe(self.notify_update)
+
+    def notify_update(self):
+        """Tell HA that the device has been updated."""
+        LOGGER.debug("Device update notification received: %s", self.name)
+        super().schedule_update_ha_state()
```

Comment:
If this is a callback from async, add `@callback` and you can write the state directly.
```suggestion
        self.async_write_ha_state()
```

Reply:
`from homeassistant.core import callback`

Reply:
Done.